lib/Test/MockTime/HiRes.pm

```diff
@@ -80,7 +80,7 @@ sub gettimeofday() {
     if (defined $Test::MockTime::fixed) {
         return wantarray ? do {
             my $int_part = int($Test::MockTime::fixed);
-            ($int_part, 1_000_000 * ($Test::MockTime::fixed - $int_part))
+            ($int_part, 1_000_000 * sprintf('%.6f', ($Test::MockTime::fixed - $int_part)))
```

Comment:
`sprintf('%.6f', $x)` rounds floating point number `$x`, and `int($x)` floors.

In a test written at https://github.com/plack/Plack/pull/551/files#diff-66e896132aa1d236dfae7cbf05485434 , after `sleep`ing 0.3 sec `tv_interval` becomes 0.2999..., not 0.3.  I guess this is due to perl's floating point number handling, and rounding by `sprintf` can avoid this problem.
```perl
# Simplified version of https://github.com/plack/Plack/pull/551/files#diff-66e896132aa1d236dfae7cbf05485434

my $t1 = [Time::HiRes::gettimeofday()];
Time::HiRes::sleep 0.3;
my $t2 = [Time::HiRes::gettimeofday()];
warn scalar Time::HiRes::tv_interval($time, $now) * 1_000_000;
```